pkg/couchdb/mango/index.go

```diff
@@ -26,9 +26,12 @@ type Index struct {
 
 // IndexOnFields constructs a new Index
 // it lets couchdb defaults for index & designdoc names.
-func IndexOnFields(doctype string, fields ...string) *Index {
+func IndexOnFields(doctype, name string, fields []string) *Index {
 	return &Index{
 		Doctype: doctype,
-		Request: &IndexRequest{Index: IndexFields(fields)},
+		Request: &IndexRequest{
+			DDoc:  name,
```

Comment:
The reason I used DDoc here is that it is how I succeeded to referer to them using the `use_index` field (the `name` did not work). I chose to name them so that we use this `use_index` parameter in our `FindDocs` requests to be more explicit about the index associated with the request.

Is there is a way to tell couchdb not to duplicate an index ?